app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCStatus.scala

```diff
@@ -1,360 +1,676 @@
+          feeRate,
+          totalCollateral,
+          localCollateral
+        )
+      case DLCState.Broadcasted =>
+        Broadcasted(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId
+        )
+      case DLCState.Confirmed =>
+        Confirmed(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId
+        )
+      case DLCState.Claimed =>
+        Claimed(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId,
+          oracleSigs,
+          outcome
+        )
+      case DLCState.RemoteClaimed =>
+        require(oracleSigs.size == 1,
+                "Remote claimed should only have one oracle sig")
+        RemoteClaimed(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId,
+          oracleSigs.head,
+          outcome
+        )
+      case DLCState.Refunded =>
+        Refunded(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId
+        )
+
     }
   }
 
-  /** The state where the remote party has successfully
-    * executed a unilateral execution by broadcasting both
-    * a CET and successfully spending their to_local output.
-    */
-  case class RemoteClaimed(
-      eventId: Sha256DigestBE,
+  def calculateOutcomeAndSig(
       isInitiator: Boolean,
       offer: DLCOffer,
       accept: DLCAccept,
       sign: DLCSign,
-      fundingTx: Transaction,
-      cet: Transaction,
-      closingTx: Transaction)
-      extends AcceptedDLCStatus {
-    override val statusString: String = "REMOTE CLAIMED"
-  }
+      cet: Transaction): (SchnorrDigitalSignature, DLCOutcomeType) = {
```

Comment:
Maybe we can just make the `cet: WitnessTransaction` as a parameter?

Reply:
Added a match instead as generally this is only called when you are in a remote-claimed situation and our node setup just gives us a `Transaction` so nice to do it for the caller here